test/Confluent.Kafka.IntegrationTests/Tests/ClosedHandle.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public static partial class Tests
+    {
+        /// <summary>
+        ///     Tests that ObjectDisposedException is thrown rather than AccessViolationException
+        ///     when Dispose has been called
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public static void Producer_ClosedHandle(string bootstrapServers, string topic, string partitionedTopic)
+        {
+            // Fix https://github.com/confluentinc/confluent-kafka-dotnet/issues/258
```

Comment:
My compiler is always complaining on no commentary on public methods, so I add them everytime now. The point was also to state that it was disposed thar was tested